sqlmesh/core/audit/definition.py

```diff
@@ -247,22 +261,154 @@ def macro_definitions(self) -> t.List[d.MacroDef]:
         """All macro definitions from the list of expressions."""
         return [s for s in self.expressions if isinstance(s, d.MacroDef)]
 
-    def _create_query_renderer(self, model: _Model) -> QueryRenderer:
+    def _create_query_renderer(
+        self, model_or_audit: t.Union[_Model, StandaloneAudit]
+    ) -> QueryRenderer:
+        kwargs: t.Dict[str, t.Any] = {}
+        if isinstance(model_or_audit, _Model):
+            kwargs["only_execution_time"] = model_or_audit.kind.only_execution_time
+
         return QueryRenderer(
             self.query,
-            self.dialect or model.dialect,
+            self.dialect or model_or_audit.dialect,
             self.macro_definitions,
             path=self._path or Path(),
             jinja_macro_registry=self.jinja_macros,
-            python_env=model.python_env,
-            only_execution_time=model.kind.only_execution_time,
+            python_env=model_or_audit.python_env,
+            **kwargs,
         )
 
 
+class StandaloneAudit(Node):
+    """
+    Args:
+        name: The name of the model, which is of the form [catalog].[db].table.
+            The catalog and db are optional.
+        owner: The owner of the model.
+        description: The optional model description.
+        stamp: An optional arbitrary string sequence used to create new model versions without making
+            changes to any of the functional components of the definition.
+        tags:
+        hash_raw_query:
+    """
+
+    name: str
+    depends_on_: t.Optional[t.Set[str]] = Field(default=None, alias="depends_on")
+    audit: Audit
+    tags: t.List[str] = []
+    hash_raw_query: bool = False
+
+    python_env_: t.Optional[t.Dict[str, Executable]] = Field(default=None, alias="python_env")
+    source_type: Literal["audit"] = "audit"
+
+    _path: Path = Path()
+    _depends_on: t.Optional[t.Set[str]] = None
+
+    @property
+    def depends_on(self) -> t.Set[str]:
+        if self._depends_on is None:
+            self._depends_on = self.depends_on_ or set()
+
+            query = exp.select("*").from_("table")  # self.render_query(optimize=False)
```

Comment:
What is this for? Why are we doing this?

Reply:
Temporary code. Needs to be fixed.